src/Symfony/Component/Serializer/Tests/Fixtures/NullableOptionalConstructorArgumentDummy.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Tests\Fixtures;
+
+class NullableOptionalConstructorArgumentDummy
+{
+    private $foo;
+
+    public function __construct(?\stdClass $foo = null)
```

Comment:
please remove the leading `?`, per our current codestyle

Reply:
The whole point is to test it works with `?`.

Reply:
`?` doesn't matter if the argument defaults to null, from the pov of reflection.
Don't you want to add a test case that has a nullable argument with *no default* instead?

Reply:
I added it, actually, there was one test, now there are 4 which tests the full matrix, with a default, with a value, with no default, with no value, with a default with no value, with no default, with a value.